cmd/snap-update-ns/system.go

```diff
@@ -66,9 +66,13 @@ func (upCtx *SystemProfileUpdateContext) Assumptions() *Assumptions {
 	//
 	// /snap/$SNAP_INSTANCE_NAME and /snap/$SNAP_NAME are added to allow
 	// remapping for parallel installs only when the snap has an instance key
+	//
+	// /var/lib/snapd/hostfs is there to allow the content interface to bridge
+	// to arbitrary directories from the host, even if they don't exist (they
+	// are created on the host in such case).
 	as := &Assumptions{}
 	instanceName := upCtx.InstanceName()
-	as.AddUnrestrictedPaths("/tmp", "/var/snap", "/snap/"+instanceName)
+	as.AddUnrestrictedPaths("/tmp", "/var/snap", "/snap/"+instanceName, "/var/lib/snapd/hostfs")
```

Comment:
Should this be limited to /var/lib/snapd/hosts/var/lib/dhcp?

Wondering if we could somehow convey that as part of the mount profile? maybe use a special option for that, eg.
```
/var/lib/snapd/hostfs/var/lib/dhcp none none x-snapd.unrestricted=true 0 0
```

Reply:
I think it could be but that would be redundant.

This is only exercised when an interface asks to use it, you can think that an interface is the declaration. We already have to say the same thing twice (osutil.MountEntry and apparmor spec.AddUpdateNS) so I think this is enough.

Reply:
I would actually limit this to /var/lib/snapd/hostfs/var at this level until we have other use cases

Reply:
Done